openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -172,6 +172,52 @@ def test_get_cohort_id(self):
             lambda: cohorts.get_cohort_id(user, SlashSeparatedCourseKey("course", "does_not", "exist"))
         )
 
+    def test_assignment_type(self):
+        """
+        Make sure that cohorts.set_assignment_type() and cohorts.get_assignment_type() works correctly.
+        """
+        course = modulestore().get_course(self.toy_course_key)
+        self.assertFalse(course.is_cohorted)
+
+        config_course_cohorts(course, discussions=[], cohorted=True)
+
+        # We are creating two random cohorts because we can't change assignment type of
+        # random cohort if it is the only random cohort present.
+        cohort1 = CohortFactory(course_id=course.id, name="TestCohort1")
+        CourseCohortFactory(course_user_group=cohort1, assignment_type=CourseCohort.RANDOM)
+        cohort2 = CohortFactory(course_id=course.id, name="TestCohort2")
+        CourseCohortFactory(course_user_group=cohort2, assignment_type=CourseCohort.RANDOM)
+        cohort3 = CohortFactory(course_id=course.id, name="TestCohort3")
+        CourseCohortFactory(course_user_group=cohort3, assignment_type=CourseCohort.MANUAL)
```

Comment:
would be good to find a way to create a testing cohort with a single call rather than two, to make it more readable / less error prone.